courses/machine_learning/deepdive2/image_classification/solutions/3_tf_hub_transfer_learning.ipynb

```diff
@@ -0,0 +1,750 @@
+{
```

Comment:
<p>We can use&nbsp;<a href="https://www.tensorflow.org/api_docs/python/tf/data/Dataset" target="_blank">tf.data.datasets</a>&nbsp;to build a <strong>iterable</strong> based on files in a Google Cloud Storage Bucket.</p><p> </p><p>The&nbsp;<code>Dataset</code>&nbsp;object is a Python iterable.</p><p> </p><p>As an aside, <code>ImageDataGenerator </code> can be used with tf.data, but it looks like it's still slow because of Python GIL. just fyi :)</p><p> </p><p><a href="https://www.tensorflow.org/guide/data#consuming_python_generators" target="_blank">https://www.tensorflow.org/guide/data#consuming_python_generators</a></p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/training-data-analyst/pull/662/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='8'/>

Reply:
<p>Ah, thanks Chris, I always get tripped up on the difference between the two :)</p>